### PR TITLE
Fixed dates on yield curve functions, test functions and doc strings

### PR DESCRIPTION
**Pull request by pmpmchaves**
# Description

Hello, this is my first pull request on the platform :) Hope it helps, let me know if you have any questions!

Fixes Bug issue: https://github.com/OpenBB-finance/OpenBBTerminal/issues/3615

- [X] Summary of the change / bug fix: 
In the SDK, when calling the function openbb.economy.ycrv(), the argument date only accepts a datetime date.
For example, it only works when converting the input into a datetime element:
get_yield_curve(datetime(2022,11,1))
How it should work:
get_yield_curve("2022-11-01")
- [x] Link # issue, if applicable: https://github.com/OpenBB-finance/OpenBBTerminal/issues/3615
- [x] Screenshot of the feature or the bug before/after fix, if applicable:
image

- [x] Relevant motivation and context:
- [x] List any dependencies that are required for this change:

# How has this been tested?
On the OpenBB terminal:
`economy/ycrv`
`economy/ycrv -d 2022-05-23`

On the SDK (Jupyter Notebook):
`from openbb_terminal.sdk import openbb`
`openbb.economy.ycrv('2022-05-23')`
`openbb.economy.ycrv_chart('2022-05-23')`

- [x] Make sure affected commands still run in terminal:
- [x] Ensure the SDK still works
- [x] Check any related reports

# Checklist:

- [x] Update [our Hugo documentation](https://openbb-finance.github.io/OpenBBTerminal/) following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/website).
- [x] Update our tests following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/tests).
- [x] Make sure you are following our [CONTRIBUTING guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/blob/main/CONTRIBUTING.md).
- [x] If a feature was added make sure to add it to the corresponding [scripts file](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/scripts).

# Others
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.

### CODE REVIEW

**Comment by hjoaquim** [comment]
Thanks for this contribution and welcome :rocket: !! Nice catch here :sunglasses:

File: openbb_terminal/economy/economy_controller.py [hunk 1]
```diff
@@ -1018,7 +1018,6 @@ def call_ycrv(self, other_args: List[str]):
         parser.add_argument(
             "-d",
             "--date",
-            type=valid_date,
```
Comment (hjoaquim) [inline openbb_terminal/economy/economy_controller.py:1021]: I would keep this in order to ensure it's still a valid date, and then have `ns_parser.date.strftime("%Y-%m-%d")` passing the argument to `fred_view.display_yield_curve`. WDYT?

**Review by hjoaquim — CHANGES_REQUESTED** [review]

**Review by hjoaquim — APPROVED** [review]

**Comment by hjoaquim** [comment]
@pmpmchaves I'm sorry to step ahead, we're trying to get rid of many bugs as possible before the release!

**Comment by pmpmchaves** [comment]
> @pmpmchaves I'm sorry to step ahead, we're trying to get rid of as many bugs as possible before the release!

Not a problem @hjoaquim ! Happy to help :)

> Thanks for this contribution and welcome @pmpmchaves 🚀 !! Nice catch here 😎

You are very welcome, keep up the great work guys 🚀

**Pull request merged**